### ISSUE TITLE
[RLMObject objectForPrimaryKey:] verifyThread exception

### ISSUE THREAD

**Issue by jpstern** [realm/realm-cocoa#6693]
## Goals
I hope to achieve a crash free Realm experience

## Expected Results
I expect Realm to use a local thread bound Realm when looking up an object by it's primary key

## Actual Results
```
Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Triggered by Thread:  20

Last Exception Backtrace:
Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Triggered by Thread:  20

Last Exception Backtrace:
0   CoreFoundation                	0x18cba2300 __exceptionPreprocess + 228 (NSException.m:199)
1   libobjc.A.dylib               	0x18c8b6c1c objc_exception_throw + 60 (objc-exception.mm:565)
2   Realm                         	0x101cc8898 -[RLMRealm verifyThread] + 72
3   Realm                         	0x101c64da4 RLMGetObject + 92
4   Realm                         	0x101c5b4a8 +[RLMObject objectForPrimaryKey:] + 108
5   --                   	         0x100b60270 UpdateMapOperation.main() + 668 
```

## Steps for others to Reproduce
1) Create a custom subclass of `Operation`
2) Initialize that that subclass with the primaryKey value
3) Inside the function main, lookup the object by it's primary key
4) Occassionally random crash

## Code Sample
```
class CrashOperation: Operation {
  let value: String
  init(_ string: String) {
     self.value = string
     super.init()
  }
  override func main() {
    let object = Object..object(forPrimaryKey: self.value) //CRASH
    ...
  }
}

let op = CrashOperation("1234")
operationQueue.addOperation(op)
```

This also occurs using global dispatch queues within an autoreleasepool
```
class func asyncWrite(qos: DispatchQoS.QoSClass = .default,
                          closure: @escaping ((_ realm: RLMRealm)->Void)) {
        DispatchQueue.global(qos: qos).async {
            autoreleasepool {
                let realm = RLMRealm.default()
                _ = try? realm.transaction {
                    closure(realm)
                }
            }
        }
    }

func doSomething(id: String) {
  asyncWrite { (realm) in 
     if let object = Object.object(forPrimaryKey: id) { //CRASH
           object.dog = "cat"
           realm.addOrUpdate(object)
     }
  } 
}
```

## Version of Realm and Tooling

Realm framework version: 5.0.3 and above.  Does not appear to happen on 4.3.2

Realm Object Server version: N/A

Xcode version: 11.6

iOS/OSX version: 12.4.8, 13.5.1, 13.5.6, 14.0.0, 

Dependency manager + version: CocoaPods 1.9.3

**Comment by apptekstudios** [comment]
Possibly related to #6659

**Comment by tgoyne** [comment]
Closing this as it is probably fixed in 5.3.5. Please let us know if it is not.
